routes/preview.js

```diff
@@ -104,42 +104,17 @@ const _handlePreviewRequest = async (req, res) => {
 
   const u = url.parse(req.url, true);
   const spec = (() => {
-    const match = u.pathname.match(/^\/\[([^\]]+)\.([^\].]+)\]\/([^\.]+)\.(.+)$/);
-    if (match) {
-      const url = match[1] + '.' + match[2];
-      const hash = match[1];
-      const ext = match[2].toLowerCase();
-      const type = match[4].toLowerCase();
-      const width = match[3]?.match(/(?<=\/)[\w+.-]+.+?(?=x)/)?.[0];
-      const height = match[3]?.match(/(?<=x)[\w+.-]+/)?.[0];
-      return {
-        url,
-        hash,
-        ext,
-        type,          
-        width,
-        height
-      };
-    } else {
-      const match = u.pathname.match(/^\/([^\.]+)\.([^\/]+)\/([^\.]+)\.(.+)$/);
-      if (match) {
-        const hash = match[1];
-        const ext = match[2].toLowerCase();
-        const type = match[4].toLowerCase();
-        const url = `${storageHost}/ipfs/${hash}`;
-        const width = match[3]?.match(/(?<=\/)[\w+.-]+.+?(?=x)/)?.[0];
-        const height = match[3]?.match(/(?<=x)[\w+.-]+/)?.[0];
-        return {
-          url,
-          hash,
-          ext,
-          type,
-          width,
-          height
-        };
-      } else {
-        return null;
-      }
+
+    let {
+      url, hash, ext, type, height, width
+    } = parseQuery(u.search);
+
+    if(!url){

```

Comment:
Can you make sure there is a `hash` in this case, and then error or something?

I'm sure someone will call without `url` or `hash` and then it will try to fetch some weirdness. Better to just error now.

Reply:
Added a return null incase there is no url or hash